src/StackExchange.Redis/Enums/RedisCommand.cs

```diff
@@ -1,236 +1,465 @@
+            case RedisCommand.ZREM:
+            case RedisCommand.ZREMRANGEBYLEX:
+            case RedisCommand.ZREMRANGEBYRANK:
+            case RedisCommand.ZREMRANGEBYSCORE:
+            case RedisCommand.ZUNIONSTORE:
+                return true;
+            // Commands that can be issued anywhere
+            case RedisCommand.NONE:
+            case RedisCommand.ASKING:
+            case RedisCommand.AUTH:
+            case RedisCommand.BGREWRITEAOF:
+            case RedisCommand.BGSAVE:
+            case RedisCommand.BITCOUNT:
+            case RedisCommand.BITPOS:
+            case RedisCommand.CLIENT:
+            case RedisCommand.CLUSTER:
+            case RedisCommand.CONFIG:
+            case RedisCommand.DBSIZE:
+            case RedisCommand.DEBUG:
+            case RedisCommand.DISCARD:
+            case RedisCommand.DUMP:
+            case RedisCommand.ECHO:
+            case RedisCommand.EVAL:
+            case RedisCommand.EVALSHA:
+            case RedisCommand.EXEC:
+            case RedisCommand.EXISTS:
+            case RedisCommand.GEODIST:
+            case RedisCommand.GEOHASH:
+            case RedisCommand.GEOPOS:
+            case RedisCommand.GEORADIUS:
+            case RedisCommand.GEORADIUSBYMEMBER:
+            case RedisCommand.GEOSEARCH:
+            case RedisCommand.GET:
+            case RedisCommand.GETBIT:
+            case RedisCommand.GETRANGE:
+            case RedisCommand.HEXISTS:
+            case RedisCommand.HGET:
+            case RedisCommand.HGETALL:
+            case RedisCommand.HKEYS:
+            case RedisCommand.HLEN:
+            case RedisCommand.HMGET:
+            case RedisCommand.HRANDFIELD:
+            case RedisCommand.HSCAN:
+            case RedisCommand.HSTRLEN:
+            case RedisCommand.HVALS:
+            case RedisCommand.INFO:
+            case RedisCommand.KEYS:
+            case RedisCommand.LASTSAVE:
+            case RedisCommand.LATENCY:
+            case RedisCommand.LINDEX:
+            case RedisCommand.LLEN:
+            case RedisCommand.LPOS:
+            case RedisCommand.LRANGE:
+            case RedisCommand.MEMORY:
+            case RedisCommand.MGET:
+            case RedisCommand.MONITOR:
+            case RedisCommand.MULTI:
+            case RedisCommand.OBJECT:
+            case RedisCommand.PFCOUNT:
+            case RedisCommand.PING:
+            case RedisCommand.PSUBSCRIBE:
+            case RedisCommand.PTTL:
+            case RedisCommand.PUBLISH:
+            case RedisCommand.PUBSUB:
+            case RedisCommand.PUNSUBSCRIBE:
+            case RedisCommand.QUIT:
+            case RedisCommand.RANDOMKEY:
+            case RedisCommand.READONLY:
+            case RedisCommand.READWRITE:
+            case RedisCommand.REPLICAOF:
+            case RedisCommand.ROLE:
+            case RedisCommand.SAVE:
+            case RedisCommand.SCAN:
+            case RedisCommand.SCARD:
+            case RedisCommand.SCRIPT:
```

Comment:
Hmmm I wonder if this would be considered a write command?

Reply:
Could be either, so we should leave it in this case...should clarify the comment on these are things that would 100% fail on a read-only sever for any use case, anything else should be `false` - thoughts?

Reply:
SCRIPT is definitely not a writeonly command, which is interesting because for things like `SCRIPT LOAD` you'd think it have to be propagated to the replica's?